src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementArray;
+    private int size;
+
+    public ArrayList() {
+        elementArray = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementArray.length) {
+            //elementArray = grow(elementArray, 1);
+            elementArray = grow();
+        }
+        elementArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndexEqualsSize(index);
+        if (size == elementArray.length) {
+            elementArray = grow();
+        }
```

Comment:
```suggestion
        if (size == elementArray.length) {
            elementArray = grow();
        }
        checkIndex(index);        
```
this way you can wouldn't need the second checker method